src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,28 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let devpath = args[1].clone();
```

Comment:
@bgurney-rh: Other than where we see ```unwrap()``` this use of the index operator is the only place where the program can just fail. Try invoking and passing no arguments, and you'll see an immediate failure.

What that means, of course, is that about 1/2 the lines in this ```main()``` method can be reasonably expected to cause a failure. I'll be giving you a plan for how to address this.